controllers/kubeadmconfig_controller.go

```diff
@@ -152,6 +158,8 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, re
 		initdata, err := json.Marshal(config.Spec.InitConfiguration)
 		if err != nil {
 			log.Error(err, "failed to marshal init configuration")
+			// release the lock
```

Comment:
I don't think we need these comments at every initLocker.Release call site. If you think `initLocker.Release` is not clear enough we should consider renaming it to something more clear

Reply:
I think the function name is clear enough. I will remove the comments.